In [48]:
import pandas as pd
import statsmodels.api as sm
import matplotlib.pyplot as plt
%matplotlib inline

In [49]:
df=pd.read_csv('C:/Users/Ecem/Desktop/BTS/BTS_Dersler/BTS_DataScience/Session_2/2_titanic_dataset.csv')
df.head(100)

PassengerId  Survived  Pclass  \
0             1         0       3   
1             2         1       1   
2             3         1       3   
3             4         1       1   
4             5         0       3   
5             6         0       3   
6             7         0       1   
7             8         0       3   
8             9         1       3   
9            10         1       2   
10           11         1       3   
11           12         1       1   
12           13         0       3   
13           14         0       3   
14           15         0       3   
15           16         1       2   
16           17         0       3   
17           18         1       2   
18           19         0       3   
19           20         1       3   
20           21         0       2   
21           22         1       2   
22           23         1       3   
23           24         1       1   
24           25         0       3   
25           26         1       3   
26           27         0       3   
27           28         0       1   
28           29         1       3   
29           30         0       3   
..          ...       ...     ...   
70           71         0       2   
71           72         0       3   
72           73         0       2   
73           74         0       3   
74           75         1       3   
75           76         0       3   
76           77         0       3   
77           78         0       3   
78           79         1       2   
79           80         1       3   
80           81         0       3   
81           82         1       3   
82           83         1       3   
83           84         0       1   
84           85         1       2   
85           86         1       3   
86           87         0       3   
87           88         0       3   
88           89         1       1   
89           90         0       3   
90           91         0       3   
91           92         0       3   
92           93         0       1   
93           94         0       3   
94           95         0       3   
95           96         0       3   
96           97         0       1   
97           98         1       1   
98           99         1       2   
99          100         0       2   

                                                 Name     Sex    Age  SibSp  \
0                             Braund, Mr. Owen Harris    male  22.00      1   
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.00      1   
2                              Heikkinen, Miss. Laina  female  26.00      0   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.00      1   
4                            Allen, Mr. William Henry    male  35.00      0   
5                                    Moran, Mr. James    male    NaN      0   
6                             McCarthy, Mr. Timothy J    male  54.00      0   
7                      Palsson, Master. Gosta Leonard    male   2.00      3   
8   Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.00      0   
9                 Nasser, Mrs. Nicholas (Adele Achem)  female  14.00      1   
10                    Sandstrom, Miss. Marguerite Rut  female   4.00      1   
11                           Bonnell, Miss. Elizabeth  female  58.00      0   
12                     Saundercock, Mr. William Henry    male  20.00      0   
13                        Andersson, Mr. Anders Johan    male  39.00      1   
14               Vestrom, Miss. Hulda Amanda Adolfina  female  14.00      0   
15                   Hewlett, Mrs. (Mary D Kingcome)   female  55.00      0   
16                               Rice, Master. Eugene    male   2.00      4   
17                       Williams, Mr. Charles Eugene    male    NaN      0   
18  Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.00      1   
19                            Masselmani, Mrs. Fatima  female    NaN      0   
20                               Fynney, Mr. Josep

In [50]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [51]:
total_row_count=df.shape[0]
total_row_count

891

In [52]:
##null count of age
null_age=sum(pd.isnull(df['Age']))

In [53]:
null_age

177

In [54]:
total_row_count

891

In [55]:
##Frequency of null charcter for Age

FNA=(null_age/total_row_count)*100
FNA

19.865319865319865

In [56]:
df['Age'].mean()

29.69911764705882

In [58]:
## We should null character mean or median for FNA grather than % 19
df['Age'] = df['Age'].fillna((df['Age'].mean()))

In [59]:
#Checking null character of Age
check_null_age=sum(pd.isnull(df['Age']))
check_null_age

0

## Multiple Regression without constant

In [80]:
x=df[['Age','Fare']]
y=df['Survived']

In [81]:
model=sm.OLS(y,x).fit()

In [82]:
model.params

Age     0.007368
Fare    0.003166
dtype: float64

In [83]:
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:               Survived   R-squared:                       0.364
Model:                            OLS   Adj. R-squared:                  0.362
Method:                 Least Squares   F-statistic:                     253.9
Date:                Sun, 29 Oct 2017   Prob (F-statistic):           5.79e-88
Time:                        13:17:31   Log-Likelihood:                -636.37
No. Observations:                 891   AIC:                             1277.
Df Residuals:                     889   BIC:                             1286.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Age            0.0074      0.001     12.226      0.000       0.006       0.009
Fare           0.0032      0.000      9.593      0.000       0.003       0.004
==============================================================================
Omnibus:                      983.402   Durbin-Watson:                   1.940
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               87.563
Skew:                           0.449   Prob(JB):                     9.68e-20
Kurtosis:                       1.754   Cond. No.                         2.38
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The model have R-squared=0.364. All of variables that using model p value <0.05 and we accept our hypotesis.Age and fare affet the survived 

In [84]:
prediction=model.predict(x)
prediction

0      0.185054
1      0.505682
2      0.216663
3      0.426007
4      0.283373
5      0.245608
6      0.562084
7      0.081463
8      0.234190
9      0.198362
10     0.082347
11     0.511414
12     0.172850
13     0.386379
14     0.128022
15     0.455906
16     0.106950
17     0.259987
18     0.285403
19     0.241703
20     0.340205
21     0.291677
22     0.135944
23     0.318706
24     0.125671
25     0.379367
26     0.241703
27     0.972688
28     0.243774
29     0.243827
         ...   
861    0.191142
862    0.435766
863    0.439032
864    0.217995
865    0.350622
866    0.242817
867    0.388289
868    0.248906
869    0.064722
870    0.216571
871    0.512697
872    0.258980
873    0.374798
874    0.282295
875    0.133398
876    0.178536
877    0.164994
878    0.243827
879    0.675907
880    0.266523
881    0.268148
882    0.195397
883    0.239553
884    0.206525
885    0.379571
886    0.240100
887    0.234979
888    0.293073
889    0.286556
890    0.260318
Length: 891, dtype: floa

## Multiple Regression with constant

In [85]:
y1=df['Survived']
x1=df[['Age','Fare']]

In [86]:
x1=sm.add_constant(x)

In [87]:
model1=sm.OLS(y1,x1).fit()

In [88]:
model1.params

const    0.404635
Age     -0.003524
Fare     0.002604
dtype: float64

In [89]:
prediction1=model1.predict(x1)
prediction1

0      0.345989
1      0.456347
2      0.333652
3      0.419570
4      0.302263
5      0.322005
6      0.349395
7      0.452465
8      0.338482
9      0.433604
10     0.434026
11     0.269388
12     0.355120
13     0.348644
14     0.375753
15     0.252488
16     0.473427
17     0.333832
18     0.342267
19     0.318794
20     0.349004
21     0.318676
22     0.372685
23     0.398408
24     0.431322
25     0.352461
26     0.318794
27     1.022518
28     0.320498
29     0.320541
         ...   
861    0.360580
862    0.303010
863    0.481085
864    0.353914
865    0.290486
866    0.345578
867    0.426885
868    0.324718
869    0.419530
870    0.333576
871    0.375863
872    0.301369
873    0.262451
874    0.368463
875    0.370591
876    0.359796
877    0.358243
878    0.320541
879    0.423840
880    0.384242
881    0.308909
882    0.354496
883    0.333309
884    0.334897
885    0.343046
886    0.343343
887    0.415801
888    0.361043
889    0.391134
890    0.312053
Length: 891, dtype: floa

In [90]:
model1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:               Survived   R-squared:                       0.075
Model:                            OLS   Adj. R-squared:                  0.073
Method:                 Least Squares   F-statistic:                     36.00
Date:                Sun, 29 Oct 2017   Prob (F-statistic):           9.28e-16
Time:                        13:17:43   Log-Likelihood:                -587.23
No. Observations:                 891   AIC:                             1180.
Df Residuals:                     888   BIC:                             1195.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.4046      0.040     10.176      0.000       0.327       0.483
Age           -0.0035      0.001     -2.905      0.004      -0.006      -0.001
Fare           0.0026      0.000      8.205      0.000       0.002       0.003
==============================================================================
Omnibus:                       31.268   Durbin-Watson:                   1.965
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              121.041
Skew:                           0.478   Prob(JB):                     5.20e-27
Kurtosis:                       1.469   Cond. No.                         158.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The model have R-squared=0.075. All of variables that using model p value <0.05 and we accept our hypotesis.
Age and fare affet the survived.
